core/src/com/unciv/logic/civilization/diplomacy/CityStateFunctions.kt

```diff
@@ -540,6 +540,17 @@ class CityStateFunctions(val civInfo: Civilization) {
                     diplomacy.setFlag(DiplomacyFlags.BorderConflict, 10)
                 }
             }
+            
+            // If citystate gets tile stolen from another civ, then the civ gets a diplomacy Influence debuffer.
+            if (diplomacy.hasModifier(DiplomaticModifiers.StealingTerritory) && diplomacy.isRelationshipLevelLT(RelationshipLevel.Friend)) {
+                diplomacy.addInfluence(-25f)
+                if (!diplomacy.hasFlag(DiplomacyFlags.BorderOccupation)) {
+                    otherCiv.popupAlerts.add(PopupAlert(AlertType.BorderOccupation, civInfo.civName))
+                    
+                    // to only show popup once.
+                    diplomacy.setFlag(DiplomacyFlags.BorderOccupation, -1)
```

Comment:
Flag name is not indicative of what it's actually remembering - should be "TilesStolen" or similar, this sounds like you have units on tiles on their borders